### ISSUE TITLE
opening hours in parks

### ISSUE THREAD

**Issue by emiltin** [Project-OSRM/osrm-backend#33]
perhaps this is technically difficult, but it would be good if osrm could handle opening hours.

for example, there are a number of parks in copenhagen where you're allowed to bicycle. but the parks have opening hours. so during the day, you should be routed through the park, but not when they're closed in the evening/night. for this to work, a request should include a time and a departure/arrival flag.

**Comment by TheMarex** [comment]
Time dependent routing is not possible with the current approach. The best we could do is return the opening hours with the response to make it visible to the user/application that re-routing might be needed.

**Comment by emiltin** [comment]
for some use cases you could have different profiles for day and night. for example for biking or walking this would at least capture things like parks that close or are not lit after dark.

**Comment by daniel-j-h** [comment]
Superseded by @oxidase's work in https://github.com/Project-OSRM/osrm-backend/pull/3415 and the corresponding issue https://github.com/Project-OSRM/osrm-backend/issues/3414.

Once the opening hours utility tool is in place you can at least contract, say, for every 15 minutes or hour and get different routes back.

**Comment by MichalPP** [comment]
does #4006 fix this issue? making several exclude classes based on the time (or night and day routing)

**Comment by daniel-j-h** [comment]
Yes if you pre-process the data and add conditional turn restriction for all turns going over the conditional access tag - which is a bit of a hack. Conditional access tags are ticket in: https://github.com/Project-OSRM/osrm-backend/issues/4231
